src/operator/nn/cudnn/cudnn_batch_norm-inl.h

```diff
@@ -44,6 +44,9 @@ enum CuDNNBatchNormOpAuxiliary {kMovingMean, kMovingInvVar};
 #if defined(__CUDACC__)
 template<typename DType>
 class CuDNNBatchNormOp {
+  // This file was formerly included in batch_norm.cu only when CUDNN_MAJOR >= 5
```

Comment:
This comment may be useful for review of this PR, but I don't think it makes sense to have it going forward.

Reply:
I can take this out.  Also explanations of what features of cuDNN are being used that require the assert, as you mention below.